homeassistant/components/maxcube/binary_sensor.py

```diff
@@ -20,36 +23,50 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         add_entities(devices)
 
 
-class MaxCubeShutter(BinarySensorEntity):
-    """Representation of a MAX! Cube Binary Sensor device."""
+class MaxCubeBinarySensorBase(BinarySensorEntity):
+    """Base class for maxcube binary sensors."""
 
     def __init__(self, handler, device):
         """Initialize MAX! Cube BinarySensorEntity."""
-        room = handler.cube.room_by_id(device.room_id)
-        self._name = f"{room.name} {device.name}"
         self._cubehandle = handler
         self._device = device
+        self._room = handler.cube.room_by_id(device.room_id)
+        self._attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
 
-    @property
-    def name(self):
-        """Return the name of the BinarySensorEntity."""
-        return self._name
+    def update(self):
+        """Get latest data from MAX! Cube."""
+        self._cubehandle.update()
 
-    @property
-    def unique_id(self):
-        """Return a unique ID."""
-        return self._device.serial
 
-    @property
-    def device_class(self):
-        """Return the class of this sensor."""
-        return DEVICE_CLASS_WINDOW
+class MaxCubeShutter(MaxCubeBinarySensorBase):
+    """Representation of a MAX! Cube Binary Sensor device."""
+
+    def __init__(self, handler, device):
+        """Initialize MAX! Cube BinarySensorEntity."""
+        super().__init__(handler, device)
+
+        self._attr_name = f"{self._room.name} {self._device.name}"
+        self._attr_unique_id = self._device.serial
+        self._attr_device_class = DEVICE_CLASS_WINDOW
```

Comment:
This should be a class variable

Reply:
Changed https://github.com/home-assistant/core/pull/58699/commits/4e328371499388fe6cb4db64bf746f788c797297